binary_search_tree/tree.py

```diff
@@ -14,35 +14,146 @@ class Tree:
     def __init__(self):
         self.root = None
 
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
+        
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node
+    
     # Time Complexity: 
     # Space Complexity: 
     def add(self, key, value = None):
-        pass
+
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
+
+    #Iterative solution
+    #Time complexity: 0(log n) -  if balanced
+    #Space complexity: 0(1) - will always only increase by 1 if we add
+    # if self.root == None:
+    #     self.root = TreeNode(key, value=None)
+    # else:
+    #     previous = None
+    #     current = self.root
+
+    #     while current != None:
+    #         previous = current
+    #         if key <= current.key:
+    #             current = current.left
+    #         else:
+    #             current = current.right
+        
+    #     if key <= previous.key:
+    #         parent.left == TreeNode(key, value)
+    #     else:
+    #         parent.right = TreeNode(key, value)
+
 
     # Time Complexity: 
     # Space Complexity: 
     def find(self, key):
+
+        if self.root == None:
+            return None
+
+        current = self.root
+        while current != None:
+            if current.key == key:
+                return current.value
+            elif current.key < key:
+                current = current.right
+            else:
+                current = current.left
+
+        return None
+
+
+    def inorder_helper(self):
         pass
 
     # Time Complexity: 
     # Space Complexity: 
     def inorder(self):
-        pass
+    #left, root, right
+        nodes_array = []
+        current =self.root
+
+        if self.root == None:
+            return []
 
+        while current != None:
+            nodes_array = self.inorder(current.left)
+            nodes_array.append(current.value)
+            nodes_array = nodes_array + self.inorder(current.right) 
```

Comment:
You seem to be trying to do both a recursive and a looping version of this method.  Your `inorder` method doesn't take any parameters, maybe you meant `inorder_helper` which is incomplete.